### PR TITLE
feat: support for extra field in registry type

### PR DESCRIPTION
**Pull request by ironAiken2**
### This PR resolves [#2914](https://github.com/lablup/backend.ai-webui/issues/2914) issue

related Core PR: [#3208](https://github.com/lablup/backend.ai/pull/3208)

24.09~24.09.2 : Using `container_registry_nodes` but does not support extra fields.
24.09.3 ~ : support extra field

**Changes:**
Extra field and argument are supported from version 24.09.3 onwards. 
Because the arguments for each mutation are not provided as input types, you must create a separate mutation if you want to exclude a specific argument.

- Added `extra` field to container registry nodes to support additional JSON metadata
- Added JSON editor UI component for managing extra information in container registry editor
- Added validation for proper JSON formatting with error messages
- Made `value` and `onChange` props optional in BAICodeEditor component
- Added translations for JSON format validation messages across all languages

**How to test:**
1. move to registry page
2. enter Extra Information to verify that the request succeeds only for valid JSON types.

|dark|light|
|---|---|
|||

**Checklist:** (if applicable)

- [ ] Mention to the original issue
- [ ] Documentation
- [ ] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

**Comment by ironAiken2** [comment]
* **#2927** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2927?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2927?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by yomybaby — CHANGES_REQUESTED** [review]
Please add a condition for version compatibility.
I got an error below on v24.09.2 
```
No data returned for operation `ContainerRegistryListQuery`, got error(s): Cannot query field 'extra' on type 'ContainerRegistryNode'. See the error `source` property for more information.
```

File: react/src/components/ContainerRegistryEditorModal.tsx [hunk 1]
```diff
@@ -150,37 +90,21 @@ const ContainerRegistryEditorModal: React.FC<
                 ? null // unset
                 : values.password
               : undefined, // no change
+          extra:
+            _.isEmpty(values.extra) || _.isNil(values.extra)
```
Comment (agatha197) [inline react/src/components/ContainerRegistryEditorModal.tsx]: Is `isNil` a subset of `isEmpty`?

**Review by agatha197 — APPROVED** [review]
I'm not sure why you used isEmpty and isNil together, but everything else looks good to me. Good!

File: react/src/components/ContainerRegistryEditorModal.tsx (hunk 1 above)
Comment (ironAiken2) [reply react/src/components/ContainerRegistryEditorModal.tsx]: `isNil` contains only validate for null or undefined. `isEmpty` was added to validate values such as the empty string ''.

Comment (yomybaby) [reply react/src/components/ContainerRegistryEditorModal.tsx]: When `isNil(something)` is true, isEmpty(something` is always true. please remove `isNil` part.

**Review by yomybaby — CHANGES_REQUESTED** [review]

**Review by yomybaby — APPROVED** [review]
LGTM

**Pull request merged**